### ISSUE TITLE
api.derive.session.info() shows unused/non-applicable data

### ISSUE THREAD

**Issue by jak-pan** [polkadot-js/api#1605]
Using api beta30 - getting derived session info returns wrong value on current kusama node.
Chain height currently at BN:826889 while value for last session(epoch?) change is showing value of 262453039

It looks like other data is also wrong - if we look at polkascan and look into events on kusamacc2 - session and era starts do not correspond to data shown in session info.

According to session info, there should have been session starts at blocks  824870, 822470, 820070 which is calculated from 
`await api.rpc.chain.getHeader()).number.toNumber() - sessionProgress`
 which doesn't look like it's right because 
`await api.query.system.events.at(await api.rpc.chain.getBlockHash(824870))` 
doesn't return events that should be emitted on new nession start. Whereas
`await api.query.system.events.at(await api.rpc.chain.getBlockHash(824956))` 
does return these events

it looks like eraProgress returns same value as sessionProgress which is also wrong.

https://polkascan.io/pre/kusama-cc2/event 
![Screen Shot 2019-11-26 at 01 57 40](https://user-images.githubusercontent.com/3409250/69590640-663c0a80-0ff0-11ea-940c-966209cb5aa7.png)

https://github.com/polkadot-js/api/blob/4cea617b13efa9db9addb17f6c4e31e1df025246/packages/api-derive/src/session/info.ts#L63

**Comment by jacogr** [comment]
Yes. It should actually be removed, and is on the list since it is actually a v1 era field. The lastLengthChange won’t be applicable since it is a constant.

**Comment by jacogr** [comment]
As for eraProgress, that is correct now in master and the current beta releases - it slipped through somewhere. (On Kusama they are equivalent atm, since a new era is forced with each epoch change, but you can verify on a dev chain)

**Comment by jak-pan** [comment]
There is still problem with sessionProgress as shown in this example with latest beta.36 api
I got real session starts from https://polkascan.io/pre/kusama-cc2/event?module=session&page=1

I need to get past session starts so I can work with reward/slash info without going through all of the blocks. I also noticed that real session don't last 2400 blocks but are almost always shorter, so I can't caluclate it simply by going - 2400 blocks from last session. Is there any good resource on this?

```typescript
let sessionInfo = await api.derive.session.info()
  let sessionProgress = sessionInfo.sessionProgress.toNumber()
  let currentBlockNumber = (await api.rpc.chain.getHeader()).number.toNumber()
  let sessionStart = currentBlockNumber - sessionProgress
  let events = await api.query.system.events.at(
    await api.rpc.chain.getBlockHash(sessionStart)
  )
  console.log(
    'expected epoch duration', 
    api.consts.babe.epochDuration.toNumber()
  )
  console.log(
    'expected era start from query api',
    (await api.query.staking.currentEraStart()).toNumber()
  )
  console.log(
    'Logging events for session #',
    sessionInfo.currentIndex.toNumber() - 1,
    'calculated to start at block #',
    sessionStart,
    'current era / session progress:',
    sessionProgress,
    'with last block',
    currentBlockNumber
  )
  events.forEach(record => {
    console.log(JSON.stringify(record.event.meta.documentation))
  })

let realSessionStart = 850718
  let sessionStartEvents = await api.query.system.events.at(
    await api.rpc.chain.getBlockHash(realSessionStart)
  )
  console.log(
    'Logging events for block #',
    realSessionStart,
    'with diff of',
    sessionStart - realSessionStart
  )
  sessionStartEvents.forEach(record => {
    console.log(JSON.stringify(record.event.meta.documentation))
  })
```

![Screen Shot 2019-11-27 at 21 15 07](https://user-images.githubusercontent.com/3409250/69756639-470dbc00-115b-11ea-9ddf-af379f3a3113.png)

**Comment by jacogr** [comment]
A Babe epoch is defined in terms of slots, not blocks. Depending on how long blocks take to produce, if produced blocks are primary or secondary, slots can be left open. So the start and genesis slots, do not mean blocks, neither do the length translate to blocks.

TL;DR An epoch will generally always be <2400 blocks, but will always contain 2400 slots of which some can be empty. There is no specific counter for blocks a session lasted.

(NOT: In Aura, sessions are defined in terms of blocks)

**Comment by jak-pan** [comment]
Oh! So there's no way to get blocks of session/era starts I need to manually go through each block to find one which has events I need.

**Comment by jacogr** [comment]
100%. You will need to check block for session start events.

(We used to have that functionality in Substrate 1.x chains, e.g. Alexander, however it is not available on-chain anymore since Babe does not need it to operate - on-chain state is for operational requirements)

**Comment by jak-pan** [comment]
Yes, I know it worked on 1.x that's why I was confused by this. Thanks for the explanation.
